packages/vaex-core/vaex/functions.py

```diff
@@ -2613,4 +2613,9 @@ def where(condition, x, y):
         return values
     # default callback is on numpy
     ar = np.where(condition, x, y)
+
+    # Cast to dtype
+    if dtype is not None:
+        ar = ar.astype(dtype)
+
```

Comment:
We can be more efficient and do sth like this before the where:
```if dtype is not None:
    if isinstance(x, numbers.Number):
        dtype_scalar = np.dtype(dtype)
        x = dtype_scalar.type(x)
    if isinstance(y, numbers.Number):
        dtype_scalar = np.dtype(dtype)
        x = dtype_scalar.type(x)
```
We'd need to the final `astype` to catch cases for arrays though.

Reply:
btw, good that you check for dtype is not None, because otherwise I think it will cast to float64!

Reply:
Good comments! I'll make the changes tonight. I should have taken this approach prior. Guess that's what 1wk out of town does to you. :)

Reply:
Ok, I made some changes. Let me know what you think!